src/test/java/seedu/address/model/ModelManagerTest.java

```diff
@@ -93,6 +97,20 @@ public void getFilteredPersonList_modifyList_throwsUnsupportedOperationException
         assertThrows(UnsupportedOperationException.class, () -> modelManager.getFilteredPersonList().remove(0));
     }
 
+    @Test
+    public void getNextSession_success() {
+        Person testPerson = new PersonBuilder().withName(VALID_NAME_BOB).withSession(VALID_SESSION_BOB).build();
+        AddressBook addressBook = new AddressBookBuilder().withPerson(testPerson).build();
+        UserPrefs userPrefs = new UserPrefs();
+        modelManager = new ModelManager(addressBook, userPrefs);
+        assertTrue((modelManager.getNextSession()
+                .equals(("Next Session: " + testPerson.getName() + " "
+                        + testPerson.getSessionList().sessionList.get(0))))
+                        || modelManager.getNextSession().equals(ModelManager.MESSAGE_NO_NEXT_SESSION_FOUND));
```

Comment:
There should not be two possible outcomes.

Also instead of `equals` you can just use `assertEquals`.

Reply:
hi ms joannnaaa, always lovely to hear from you, i used two possible outcomes because depending on the local time of the machine, there might not be a next session based on the current implementation, 

but as your hardworking teammate, after the change that ms @phyliciachristel has proposed, indeed there will not be two outcomes and i will edit accordingly!


With love
Johannes W. Davis

Reply:
> i used two possible outcomes because depending on the local time of the machine, there might not be a next session based on the current implementation,

doesn't seem like a good testing practice. how would you know the code is performing as expected if you're not even sure what the expected behaviour is at that moment? you could force there to always be a next session by adding one at Sunday 23:59. though it doesn't really matter since it's going to be changed

Reply:
mmm i gets, was just because i wanted my method to work and pass the test accordingly to my spec at all times hope u unds <3 but yes anw! will not be a problem after the change 😌

Reply:
btw there could still be no next session in the event that no session has been added for anyone (since it's not a compulsory parameter). If there are two outcomes you should create two diff test methods for each outcome and edit the session list etc accordingly to get the expected output.

Reply:
Makes sense tq for the suggestion @phyliciachristel will do it!! last stretch woo <3

Reply:
Good evening ms @cfyjoanna 

regarding the two outcomes thing, I realise even with the new implementation there will still be two outcomes depending on the system time now.

As such to prevent the two outcomes situation, (since its unavoidable to not call system time and the output changes depending on system time still) ill be using the 23:59 suggestion.

Ill be making another PR shortly.

Best regards with much adoration and love
Johannes

Reply:
> There should not be two possible outcomes.
> 
> Also instead of `equals` you can just use `assertEquals`.

Also yea would have used this but due to the two conditions ~

Reply:
your test code should never depend on something you can't control, it should be consistent throughout. in this case, you shouldn't depend on the actual time of the system. would suggest creating a model stub or wtv and override the get time methods and js hardcode a time its suppose to return, so it remains consistent and from there you can test what the next session is by returning different times (in your control)

Reply:
> Suppose i have `Sun 23:59` and `Mon 08:00`

Not sure why you are selecting difficult values. Sun 2359 and Mon 0000 would be more suitable. That way your test would only fail for a minute for the entire week. Either that or you could follow @phyliciachristel 's suggestion which is more suitable imo.

Let's say at a given time, you expect Outcome A, but the program produces Outcome B. But because your test accepts both Outcomes A and B, the test will pass even though it's wrong. Onus is on the coder to ensure this doesn't happen.

Reply:
> your test code should never depend on something you can't control, it should be consistent throughout. in this case, you shouldn't depend on the actual time of the system. would suggest creating a model stub or wtv and override the get time methods and js hardcode a time its suppose to return, so it remains consistent and from there you can test what the next session is by returning different times (in your control)

hmm but in this case we arent exactly testing the actual method itself hence it wouldnt be accurate testing and may be faulted?

Reply:
> > Suppose i have `Sun 23:59` and `Mon 08:00`
> 
> Not sure why you are selecting difficult values. Sun 2359 and Mon 0000 would be more suitable. That way your test would only fail for a minute for the entire week, although that isn't really good either. @phyliciachristel 's suggestion which is more suitable imo.
> 
> Let's say at a given time, you expect Outcome A, but the program produces Outcome B. But because your test accepts both Outcomes A and B, the test will pass even though it's wrong. Onus is on the coder to ensure this doesn't happen.

This makes sense too tho tbh, if my method returns the opposite outputs since two results possible, we wont exactly know whether its working exactly hmm.

Reply:
anyway this PR is no longer updated please check the other PR NextSessionUtil, js wna say rly appreicate u both thanks for discussing this w me!